c7n/resolver.py

```diff
@@ -40,12 +41,21 @@ def resolve(self, uri):
         else:
             # TODO: in the case of file: content and untrusted
             # third parties, uri would need sanitization
-            fh = urlopen(uri)
-            contents = fh.read().decode('utf-8')
-            fh.close()
+            req = Request(uri, headers={"Accept-Encoding": "gzip"})
+            response = urlopen(req)
+            contents = self.handle_content_encoding(response)
+            response.close()
+
         self.cache.save(("uri-resolver", uri), contents)
         return contents
 
+    def handle_content_encoding(self, response):
+        if response.info().get('Content-Encoding') != 'gzip':
+            return response.read().decode('utf-8')
+
+        data = zlib.decompress(response.read(), 16).decode('utf8')
```

Comment:
please use a value that supports zip or gzip, this value only supports gzip, also worth a comment when adding magic numbers.

Reply:
I have updated it with a descriptive constant:
`ZIP_OR_GZIP_HEADER_DETECT = zlib.MAX_WBITS | 32` should you 

Including a comment felt a little redundant but happy to include one should you deem it necessary. I was going to include a test for the zip handling but I am not sure how to go about it. 'Content-Encoding: zip' does not seem to be a valid header according to: 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding

Reply:
hmm.. good point re header handling, although urllib can also open files on disk file:/// was just concerned that 16 is a bit of a magic number without context, using an explicit variable name though provides enough context for a reader sans comment.